### ISSUE TITLE
A-weighting parameter

### ISSUE THREAD

**Issue by ZhuBaohe** [librosa/librosa#1273]
https://github.com/librosa/librosa/blob/a53fa56bdb6695a994008d5b6ccd0100870a6036/librosa/core/convert.py#L1347

In [Wikipedia](https://en.wikipedia.org/wiki/A-weighting), A-weighting numerator parameter is 12194, but in above code it is 12220.

Is it a bug?

**Comment by lostanlen** [comment]
Thank you for reporting, @ZhuBaohe 
Tagging @beasteers 🐝 who wrote PR #1134. can you help?

**Comment by bmcfee** [comment]
These constants pre-date #1134 , and can probably be traced back to @mattmcvicar 's chroma features.

The constants given on wikipedia (12194) are themselves approximations of the numbers given in appendix C of ANSI  S1.4-1983: https://law.resource.org/pub/us/cfr/ibr/002/ansi.s1.4.1983.pdf

Note however that the reference given on wikipedia in advance of their equations is https://www.cross-spectrum.com/audio/weighting.html , which actually uses the rounded values that we implement here.  **Blanket warning: do not trust wikipedia as an authoritative source!**

I hacked up a quick comparison of the three versions (librosa, wikipedia, and S1.4-1983), and calculated the curves for `freqs = librosa.fft_frequencies(sr=44100, n_fft=8192)`.  The resulting curves look virtually identical:
![image](https://user-images.githubusercontent.com/1190540/103551584-86d3c680-4e78-11eb-8805-4e37b66b5f99.png)

Numerically, the differences are well below what I would consider to be meaningful:
```python
>>> np.max(np.abs(a_librosa - a_wp))
0.006543453118510456

>>> np.max(np.abs(a_librosa - a_1983))
0.006302853855526536

>>> np.max(np.abs(a_wp - a_1983))
0.004890621752522861
```

So we're talking about differences on the order of 0.005 dB, which is well below the just noticeable difference at every audible frequency.  Note that the deviations are larger at higher frequencies:
```python
>>> plt.plot(freqs, a_wp - a_librosa)
```
![image](https://user-images.githubusercontent.com/1190540/103552142-707a3a80-4e79-11eb-875d-9912db564701.png)

(As a general rule, I tend to distrust any decibel measurements past the first decimal place anyway due to the limitations of floating point precision.)

TLDR: we don't exactly implement the 1983 standard, but the differences are not practically meaningful.  I don't think this is worth changing, and don't really consider it a bug.

**Comment by beasteers** [comment]
Hi! Thanks for this. 

So that PR that I did only added B, C, D, Z, and did not touch the existing code for A weighting. But I did note this in the PR mentioned:
>There is a discrepancy between the constants currently in librosa and the ones on the wiki page. In A_weighting librosa uses 12200 whereas the wiki page uses 12194. The difference tops out around 0.0085db (checked up to 2e15) so the difference is completely insignificant, but I just wanted to document it. B and C use 12194. I was tempted to change A too, but maybe some silly person is relying on 3 decimals of precision for their dBs so I held off.

So it doesn't make a difference to me if people want to change it for consistency, but it should be inconsequential in the end.

I just did some poking around and found this which may be the source of the original A weighting coefficients http://www.sengpielaudio.com/BerechnungDerBewertungsfilter.pdf

**Comment by bmcfee** [comment]
Thanks @beasteers ! (I guess we're about synchronized in processing winter break email...)

Glad we're in agreement on the practical importance of changing this. :)  When I wrote the previous post, I had forgotten that B and C use 12194 instead.  Since the difference is practically meaningless anyway, maybe we should revise the A-weighting constants for consistency?

**Comment by bmcfee** [comment]
I propose that we keep the numbers as is for the 0.8 series, but standardize in the 0.9 series.

**Comment by turian** [comment]
FWIW, Auraloss uses: https://github.com/csteinmetz1/auraloss/blob/main/auraloss/perceptual.py#L80

```
            f1 = 20.598997
            f2 = 107.65265
            f3 = 737.86223
            f4 = 12194.217
            A1000 = 1.9997
```
